src/js_native_api_types.h

```diff
@@ -1,7 +1,6 @@
 #ifndef SRC_JS_NATIVE_API_TYPES_H_
 #define SRC_JS_NATIVE_API_TYPES_H_
 
-#include <stddef.h>
```

Comment:
I think technically we can’t be sure that addons didn’t rely on us including this header – should this be semver-major/dont-land-on-v11.x?

Reply:
My first intention is apply modernize-deprecated-headers to whole src folder(see https://github.com/nodejs/node/pull/26840/files#r267804900). So I thought maybe do some cleanup first. But add `dont-land-on-v11.x` is necessary IMHO.

Reply:
I add one more commit to remove more unused include and refactor deprecated header.
This should break the test.

I am sure which is the best balance should we take:
* whether or not should we keep unused include
* should apply https://clang.llvm.org/extra/clang-tidy/checks/modernize-deprecated-headers.html to those file or just add `//NOLINT`

Also cc @refack

Reply:
I'd go with `// NOLINT` for these.
Also ping @gabrielschulhof

Reply:
In the end, only remove two `#include`. I am thinking adding them back and `// NOLINT`. Thoughts ?

Reply:
I give it a second thought, seems it doesn't worth tidy up.
So I just add `// NOLINT` to those files. 
Can you remove the `dont land` label ?
@addaleax

Reply:
> I give it a second thought, seems it doesn't worth tidy up.

Seems most reasonable.
But I'd go with the full:
```suggestion
#include <stddef.h> // NOLINT(modernize-deprecated-headers)
```
And a comment about why. Something like:
```c
// This is a public include file, and these includes have essentially
// became part of it's API.
```